### PR TITLE
Prune old protocol crutches

### PR DESCRIPTION
**Pull request by xeruf**
Very WIP, removes the old crutch which was only implemented for the TestClient and didn't even make much sense for that.

Fix: #158

### CODE REVIEW

**Review by anarchuser — COMMENTED** [review]
looks good, I just found two style questions

File: helpers/test-client/src/sc/TestClient.java [hunk 1]
```diff
@@ -305,6 +297,15 @@ protected void onObject(@NotNull ProtocolPacket message) {
     }
   }
 
+  private final int BIG_DECIMAL_SCALE = 6;
+  /** Calculates a new average value: average = oldAverage * ((#amount - 1)/ #amount) + newValue / #amount */
+  private BigDecimal updateAverage(BigDecimal oldAverage, int amount, BigDecimal newValue) {
```
[the remark this answers is not in the document]
Comment (xeruf) [reply helpers/test-client/src/sc/TestClient.java:302]: Doesn't the formula explain that though?

File: helpers/test-client/src/sc/TestClient.java [hunk 2]
```diff
@@ -305,6 +297,15 @@ protected void onObject(@NotNull ProtocolPacket message) {
...
+    BigDecimal decAmount = new BigDecimal(amount);
+    return oldAverage.multiply(decAmount.subtract(BigDecimal.ONE).divide(decAmount, BIG_DECIMAL_SCALE, BigDecimal.ROUND_HALF_UP))
```
[the remark this answers is not in the document]
Comment (xeruf) [reply helpers/test-client/src/sc/TestClient.java:305]: Noted for a follow-up PR ;)

File: helpers/test-client/src/sc/TestClient.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (anarchuser) [reply helpers/test-client/src/sc/TestClient.java:303]: implicitly yes
but you have to read through the formula to figure that out.
If I want to know whether `amount` includes or excludes the new value, I have to look through the formula, understand it and realise that the `(#amount - 1)` part means it includes the new value.

If you name it, e.g., `newAmount`, congruently to `newValue`, it should be clear that it includes the new value

**Pull request merged**